### ISSUE TITLE
[JENKINS-64612] clang-tidy warnigns are reported as gcc warnings

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/analysis-model#1377]
<p>From a quick look it looks like recent gcc warnings are formatted as: </p>

```
<file>:<line>:<column>: warning: <message> [-W<warning_command>]
```
<p>while clang-tidy warnings are formatted as:</p>

```
<file>:<line>:<column>: warning: <message> [<clang-tidy check>]
```
<p>So gcc warnings can be differentiated by the "-W".</p>

<p>My understanding of <a href="https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/Gcc4CompilerParser.java#L23" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/Gcc4CompilerParser.java#L23</a> is that it accepts lines with anything after "warning: ". Looking in <a href="https://godbolt.org/" class="external-link" target="_blank" rel="nofollow noopener">https://godbolt.org/</a> it looks like the "<span class="error">&#91;-W<warning_command>&#93;</span>" was not added until gcc 4.6, which would explain this.</p>

<p>So maybe there is a need for a new "gcc46+" "tool" in Warnings NG / parser in analysis-model?</p>

<p>Additionally, looking in godbolt, I see clang has used the exact same </p>

```
<file>:<line>:<column>: warning: <message> [-W<warning_command>]
```
<p>format since at least clang 3.0.0 (the oldest available version in godbolt). But</p>
<ul>
	<li>The clang tool also includes clang-tdy warnings</li>
	<li>In my codebase (building for multiple toolchains, mostly gcc-based but also for Android/clang) the clang tool reports 58 warnings while the gcc4 tool reports 119 warnings</li>
</ul>

<ul>
	<li>The parsers in analysis model are very different (one based on RegexpLineParser and the other on LookaheadParser) </li>
</ul>

<p>Since I mix clang and gcc builds in my Jenkins jobs, and given that the gcc and clang formats seem to have been identical at least since 2011 (gcc 4.6.0 and clang 3.0 releases) it would be nice if I could use a single "modern gcc/clang" tool to detect the warnings of both.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reddwarf94">reddwarf94</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-64612" target="_blank">clang-tidy warnigns are reported as gcc warnings</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akash_manna">akash_manna</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: analysis-model
<li><b>label(s)</b>: hacktoberfest, help-wanted
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2025-11-09T21:26:19+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 20260108-221235
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>From a quick look it looks like recent gcc warnings are formatted as: </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>&lt;file&gt;:&lt;line&gt;:&lt;column&gt;: warning: &lt;message&gt; [-W&lt;warning_command&gt;]</pre>
</div></div>
<p>while clang-tidy warnings are formatted as:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>&lt;file&gt;:&lt;line&gt;:&lt;column&gt;: warning: &lt;message&gt; [&lt;clang-tidy check&gt;]</pre>
</div></div>
<p>So gcc warnings can be differentiated by the "-W".</p>

<p>My understanding of <a href="https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/Gcc4CompilerParser.java#L23" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/Gcc4CompilerParser.java#L23</a> is that it accepts lines with anything after "warning: ". Looking in <a href="https://godbolt.org/" class="external-link" target="_blank" rel="nofollow noopener">https://godbolt.org/</a> it looks like the "<span class="error">&#91;-W&lt;warning_command&gt;&#93;</span>" was not added until gcc 4.6, which would explain this.</p>

<p>So maybe there is a need for a new "gcc46+" "tool" in Warnings NG / parser in analysis-model?</p>

<p>Additionally, looking in godbolt, I see clang has used the exact same </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>&lt;file&gt;:&lt;line&gt;:&lt;column&gt;: warning: &lt;message&gt; [-W&lt;warning_command&gt;]</pre>
</div></div>
<p>format since at least clang 3.0.0 (the oldest available version in godbolt). But</p>
<ul>
	<li>The clang tool also includes clang-tdy warnings</li>
	<li>In my codebase (building for multiple toolchains, mostly gcc-based but also for Android/clang) the clang tool reports 58 warnings while the gcc4 tool reports 119 warnings</li>
</ul>

<ul>
	<li>The parsers in analysis model are very different (one based on RegexpLineParser and the other on LookaheadParser) </li>
</ul>

<p>Since I mix clang and gcc builds in my Jenkins jobs, and given that the gcc and clang formats seem to have been identical at least since 2011 (gcc 4.6.0 and clang 3.0 releases) it would be nice if I could use a single "modern gcc/clang" tool to detect the warnings of both.</p></pre>
</details>
</details>
